apis/r/R/SOMAExperimentAxisQuery.R

```diff
@@ -304,21 +304,45 @@ SOMAExperimentAxisQuery <- R6::R6Class(
       # Construct the dimension names
       dim_names <- switch(collection,
         X = list(obs_labels, var_labels),
-        obsm = list(obs_labels, unique(tbl$soma_dim_1)$as_vector()),
-        varm = list(var_labels, unique(tbl$soma_dim_1)$as_vector()),
+        obsm = {
+          soma_dim_1 <- range(tbl$soma_dim_1$as_vector())
+          list(obs_labels, seq(min(soma_dim_1), max(soma_dim_1)))
+        },
+        varm = {
+          soma_dim_1 <- range(tbl$soma_dim_1$as_vector())
+          list(var_labels, seq(min(soma_dim_1), max(soma_dim_1)))
```

Comment:
So the concrete use case this accommodates is something like a matrix of PCs with one or more empty columns?

Reply:
Yes, or more concretely, a query where the selection of cells all have a PC/UMAP/embedding value of zero